content/5_Plat/RURQ.mdx

```diff
@@ -284,35 +447,152 @@ int main() {
 	}
 }
 ```
+</CPPSection>
+</LanguageSection>
 
-### Tutorial
+We can also use a somewhat generic template:
```

Comment:
care to uh
elaborate?

Reply:
good point lol

Reply:
added some explanation for how the template works. i think it should be pretty clear how it works assuming the reader understands how the previous segtree impls work